Compiling model...
'compile' took 2.569345 s

Training model...

Warning: Rectangle boundary_normal called on vertices. You may use PDE(..., exclusions=...) to exclude the vertices.
Warning: Rectangle boundary_normal called on vertices. You may use PDE(..., exclusions=...) to exclude the vertices.
Step      Train loss                                                      Test loss                                                       Test metric
0         [9.24e-02, 4.80e-01, 5.75e-01, 0.00e+00, 2.95e-04, 3.94e-02]    [7.86e-02, 4.80e-01, 5.75e-01, 0.00e+00, 2.95e-04, 3.94e-02]    []

Best model at step 0:
  train loss: 1.19e+00
  test loss: 1.17e+00
  test metric: []

'train' took 2.394006 s
/working_dir/src/pde.py:230: RuntimeWarning: divide by zero encountered in divide
  loss_weights = len(initial_losses) / initial_losses

Compiling model...
'compile' took 0.001189 s

Training model...

Step      Train loss                                                      Test loss                                                       Test metric
0         [6.00e+00, 6.00e+00, 6.00e+00, nan, 6.00e+00, 6.00e+00]         [5.10e+00, 6.00e+00, 6.00e+00, nan, 6.00e+00, 6.00e+00]         []

Best model at step 0:
  train loss: 1.19e+00
  test loss: 1.17e+00
  test metric: []

Epoch 1: saving model to ./tests/models/BioHeat_PINNs_date_time_20240731_162655/adam-1.pt ...

'train' took 3.864333 s

[2024-07-31 16:27:08,852][matplotlib.mathtext][INFO] - Substituting symbol L from STIXNonUnicode
[2024-07-31 16:27:08,879][matplotlib.mathtext][INFO] - Substituting symbol L from STIXNonUnicode
[2024-07-31 16:27:08,904][matplotlib.mathtext][INFO] - Substituting symbol L from STIXNonUnicode
[2024-07-31 16:27:08,930][matplotlib.mathtext][INFO] - Substituting symbol L from STIXNonUnicode
Generated Xobs shape: (2541, 5)
The generated test data are:  [[0.         0.         0.         0.00669285 0.        ]
 [0.         0.1        0.         0.00669285 0.        ]
 [0.         0.2        0.         0.00669285 0.        ]
 ...
 [1.         0.8        0.19627177 1.         1.        ]
 [1.         0.9        0.3237232  1.         1.        ]
 [1.         1.         0.41299888 1.         1.        ]]
Error executing job with overrides: []
Traceback (most recent call last):
  File "/working_dir/./src/main.py", line 54, in main
    train.single_observer(prj, cfg.run, "0", cfg)
  File "/working_dir/src/train.py", line 110, in single_observer
    metrics = plots.plot_and_metrics(mo, n_test)
  File "/working_dir/src/plots.py", line 180, in plot_and_metrics
    theta_pred = model.predict(g)
  File "/opt/conda/lib/python3.10/site-packages/deepxde/model.py", line 890, in predict
    y = self._outputs(False, x)
  File "/opt/conda/lib/python3.10/site-packages/deepxde/model.py", line 529, in _outputs
    outs = self.outputs(training, inputs)
  File "/opt/conda/lib/python3.10/site-packages/deepxde/model.py", line 288, in outputs
    return self.net(inputs)
  File "/opt/conda/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1511, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/opt/conda/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1520, in _call_impl
    return forward_call(*args, **kwargs)
  File "/opt/conda/lib/python3.10/site-packages/deepxde/nn/pytorch/fnn.py", line 43, in forward
    else self.activation(linear(x))
  File "/opt/conda/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1511, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/opt/conda/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1520, in _call_impl
    return forward_call(*args, **kwargs)
  File "/opt/conda/lib/python3.10/site-packages/torch/nn/modules/linear.py", line 116, in forward
    return F.linear(input, self.weight, self.bias)
RuntimeError: mat1 and mat2 shapes cannot be multiplied (2541x5 and 3x50)

Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.
